File: datasets/java_source-code/binary_tree/original.java
import java.util.ArrayList;

class Node {
    int value;
    Node left, right;

    public Node(int value) {
        this.value = value;
        left = right = null;
    }
}

public class original {

    // Inorder traversal that returns a list
    public static ArrayList<Integer> inorderTraversal(Node root) {
        ArrayList<Integer> result = new ArrayList<>();

        if (root == null) {
            return result;
        }

        result.addAll(inorderTraversal(root.left));
        result.add(root.value);
        result.addAll(inorderTraversal(root.right));

        return result;
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);

        System.out.println(inorderTraversal(root));
    }
}

File: server/datasets/controlled_benchmarks/python/dsa_clone_benchmarks/binary_tree/original.py
class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def inorder_traversal(root):
    if root is None:
        return []

    result = []
    result.extend(inorder_traversal(root.left))
    result.append(root.value)
    result.extend(inorder_traversal(root.right))

    return result

if __name__ == "__main__":
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)

    print(inorder_traversal(root))
